org.eclipse.buildship.core/src/main/java/org/eclipse/buildship/core/preferences/internal/BuildCommandConverter.java

```diff
@@ -0,0 +1,151 @@
+/*
+ * Copyright (c) 2017 the original author or authors.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ */
+
+package org.eclipse.buildship.core.preferences.internal;
+
+import java.io.ByteArrayInputStream;
+import java.io.StringWriter;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.transform.Transformer;
+import javax.xml.transform.TransformerFactory;
+import javax.xml.transform.dom.DOMSource;
+import javax.xml.transform.stream.StreamResult;
+
+import org.w3c.dom.Attr;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.Lists;
+import com.google.common.collect.Maps;
+
+import org.eclipse.core.resources.ICommand;
+import org.eclipse.core.resources.IProject;
+
+import org.eclipse.buildship.core.CorePlugin;
+
+/**
+ * Transforms build commands to XML format and vica versa.
+ */
+final class BuildCommandConverter {
```

Comment:
I did another round on this to see if I've overlooked something. Turns out we'd need to use reflection to call the platform serialization (see [`ModelObjectWriter`](https://github.com/eclipse/eclipse.platform.resources/blob/6e968c80e74a60d1f407c00e044016f882b94313/bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/ModelObjectWriter.java)).